From: Andy Gospodarek <gospo@redhat.com>
Subject: [RHEL5.1 Patch] - BZ 226902 - pci: include HP devices in NIC ordering patch and fix whitespace
Date: Wed, 7 Mar 2007 13:55:52 -0500
Bugzilla: 226902
Message-Id: <20070307185552.GD27615@gospo.rdu.redhat.com>
Changelog: [pci] include devices in NIC ordering patch and fix whitespace

commit f52383d395178afde66d049e176bb2c59a8c941a
Author: Andy Gospodarek <andy@greyhouse.net>

    PCI: add systems for automatic breadth-first device sorting
    
    This is an additional list of systems that exhibit the PCI device ordering
    issue that prompted the following patch:
    
    commit 6b4b78fed47e7380dfe9280b154e8b9bfcd4c86c
    Author: Matt Domsch <Matt_Domsch@dell.com>
    Date:   Fri Sep 29 15:23:23 2006 -0500
    
        PCI: optionally sort device lists breadth-first
    
    Adding these systems to the list prevents the need for the additional
    kernel command line argument.
    
    Signed-off-by: Andy Gospodarek <andy@greyhouse.net>
    Cc: Matt Domsch <Matt_Domsch@dell.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


--- linux-2.6.18.x86_64/arch/i386/pci/common.c.gospo	2007-03-07 11:11:59.312808000 -0500
+++ linux-2.6.18.x86_64/arch/i386/pci/common.c	2007-03-07 11:19:40.439220000 -0500
@@ -190,38 +190,126 @@ static struct dmi_system_id __devinitdat
 		},
 	},
 #endif
-        {
-                .callback = set_bf_sort,
-                .ident = "Dell PowerEdge 1950",
-                .matches = {
-                        DMI_MATCH(DMI_SYS_VENDOR, "Dell"),
-                        DMI_MATCH(DMI_PRODUCT_NAME, "PowerEdge 1950"),
-                },
-        },
-        {
-                .callback = set_bf_sort,
-                .ident = "Dell PowerEdge 1955",
-                .matches = {
-                        DMI_MATCH(DMI_SYS_VENDOR, "Dell"),
-                        DMI_MATCH(DMI_PRODUCT_NAME, "PowerEdge 1955"),
-                },
-        },
-        {
-                .callback = set_bf_sort,
-                .ident = "Dell PowerEdge 2900",
-                .matches = {
-                        DMI_MATCH(DMI_SYS_VENDOR, "Dell"),
-                        DMI_MATCH(DMI_PRODUCT_NAME, "PowerEdge 2900"),
-                },
-        },
-        {
-                .callback = set_bf_sort,
-                .ident = "Dell PowerEdge 2950",
-                .matches = {
-                        DMI_MATCH(DMI_SYS_VENDOR, "Dell"),
-                        DMI_MATCH(DMI_PRODUCT_NAME, "PowerEdge 2950"),
-                },
-        },
+	{
+		.callback = set_bf_sort,
+		.ident = "Dell PowerEdge 1950",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "PowerEdge 1950"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "Dell PowerEdge 1955",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "PowerEdge 1955"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "Dell PowerEdge 2900",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "PowerEdge 2900"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "Dell PowerEdge 2950",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "PowerEdge 2950"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL20p G3",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL20p G3"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL20p G4",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL20p G4"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL30p G1",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL30p G1"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL25p G1",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL25p G1"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL35p G1",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL35p G1"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL45p G1",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL45p G1"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL45p G2",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL45p G2"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL460c G1",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL460c G1"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL465c G1",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL465c G1"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL480c G1",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL480c G1"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant BL685c G1",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant BL685c G1"),
+		},
+	},
 	{}
 };
 

